site/_posts/2021-09-16-how-to-authenticate-your-react-app.md

```diff
@@ -0,0 +1,1087 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Greeting from './components/Greeting.js'
+const config = require('./config');
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: 'test@fusionauth.io'
+    };
+  }
+
+  render() {
+    return (
+      <div id='App'>
+        <header>
+          <h1>FusionAuth Example: React</h1>
+          <Greeting email={this.state.email}/>
+        </header>
+      </div>
+    );
+  }
+}
+
+ReactDOM.render(<App/>, document.querySelector('#root'));
+```
+
+If the user is logged in, a welcome message displays. If not, then a 'You're not logged in' message displays.  Go ahead and comment out `email: 'test@fusionauth.io'` in the
+
+`this.state = {
+    email: 'test@fusionauth.io'
+  };`
+
+block. Now, confirm a 'You're not logged in' message appears.
+
+Now, let's set the state of this email based on a call to our Express server.
+
+First, we need to set up that server!
+
+## Create the Express server
+
+*"Wait", you may say, "why do I need an Express server again?"*
+
+Again, a good question. This has to do with the Authorization Code grant. This grant relies on a user interacting with a browser, but still needs a server-side component to handle passing tokens and authorization codes to and from the authorization server, aka FusionAuth.  This keeps all of your sensitive access objects on the server-side of your app, and makes them far less likely to be intercepted and hacked.  If you want to learn more about the nuts and bolts of how the Authorization Code grant works, check out the [Modern Guide to OAuth](/learn/expert-advice/oauth/modern-guide-to-oauth/).
+
+### Configure the folders and packages
+
+Now, you need to add a package.json in your `server` folder. Do that by navigating to the `server` folder and running
+
+`npm init -y`
+
+In your `server/package.json`, add a rule that tells our app where to find our Express server.
+
+```js
+...
+"scripts": {
+  "start": "node index.js"
+},
+...
+```
+
+Then, install your dependencies.
+
+```
+npm install express cors express-session request
+```
+
+What did you just install?
+- `cors`, so that you can avoid CORS errors.
+- `express`, so that you can use the Express package.
+- `express-session`, so that you can save data in a server-side session.
+- `request`, so that you can format HTTP requests in an organized fashion
+
+Your completed `package.json` should look like this:
+
+```
+{
+  "name": "server",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "start": "node index.js"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "cors": "^2.8.5",
+    "express": "^4.17.1",
+    "express-session": "^1.17.2",
+    "request": "^2.88.2"
+  }
+}
+```
+
+### Write the Express server
+
+We need to create some routes for our Express server to use.  
+
+Create a `routes` folder in `reactauthapp/server`
+
+```bash
```

Comment:
What directory am I in here? I think I am already in `reactauthapp/server` so running this command will create too many subdirectories, correct?